### PR TITLE
[BD-19] Transition to the new Elasticsearch libs version for edx-analitics-data-api

### PR DESCRIPTION
**Pull request by Golub-Sergey**
**Description:** updates versions of the ES and the ES DSL libs from V1.5.2 to V7.8.0

**Was done:**
* update ES libs version;
* update ES libs version;
* rewrite index documents for compatibility with ES7;
* update make commands;
* update travis ci.

**Tests coverage total: 97%**

**New `roster` index mapping:**
```json
{
    "roster_test": {
        "mappings": {
            "properties": {
                "attempt_ratio_order": {
                    "type": "short"
                },
                "city": {
                    "type": "keyword"
                },
                "cohort": {
                    "type": "keyword"
                },
                "country": {
                    "type": "keyword"
                },
                "course_id": {
                    "type": "keyword"
                },
                "discussion_contributions": {
                    "type": "short"
                },
                "email": {
                    "type": "keyword"
                },
                "enrollment_date": {
                    "type": "date"
                },
                "enrollment_mode": {
                    "type": "keyword"
                },
                "gender": {
                    "type": "keyword"
                },
                "goals": {
                    "type": "keyword"
                },
                "language": {
                    "type": "keyword"
                },
                "last_updated": {
                    "type": "date"
                },
                "level_of_education": {
                    "type": "keyword"
                },
                "location": {
                    "type": "keyword"
                },
                "mailing_address": {
                    "type": "keyword"
                },
                "name": {
                    "type": "keyword"
                },
                "problem_attempts_per_completed": {
                    "type": "float"
                },
                "problems_attempted": {
                    "type": "short"
                },
                "problems_completed": {
                    "type": "short"
                },
                "segments": {
                    "type": "keyword"
                },
                "user_id": {
                    "type": "integer"
                },
                "username": {
                    "type": "keyword"
                },
                "videos_viewed": {
                    "type": "short"
                },
                "year_of_birth": {
                    "type": "short"
                }
            }
        }
    }
}
```

**Reviewers:**
@mikix
@dianakhuang

**Post merge:**
* Delete working branch

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @Golub-Sergey! I've created [BLENDED-446](https://openedx.atlassian.net/browse/BLENDED-446) to keep track of it in Jira.  More details are on the [BD-19](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596162490/BD-19+BD-20+Elasticsearch+Haystack+Upgrades) project page.

When this pull request is ready, tag your edX technical lead.

File: requirements/pip_tools.in [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Dependencies to run compile tools
-pip-tools<5.0.0                           # Contains pip-compile, used to generate pip requirements files
+pip-tools<=5.2.1                          # Contains pip-compile, used to generate pip requirements files
```
Comment (dianakhuang) [inline requirements/pip_tools.in:2]: A nitpick about this, but I think we should just remove the constraint on this requirement if it's not necessary any longer.

Comment (Golub-Sergey) [reply requirements/pip_tools.in]: Done, current constraint was removed.

**Comment by Golub-Sergey** [comment]
branch was rebased for resolving conflicts

**Comment by estute** [comment]
Hi Sergey, I can get to this PR review by end of week, hopefully beforehand. Just wanted to keep you updated.

**Comment by Golub-Sergey** [comment]
> Hi Sergey, I can get to this PR review by end of week, hopefully beforehand. Just wanted to keep you updated.
@estute , thanks

File: Makefile [hunk 2]
```diff
@@ -15,13 +13,11 @@ requirements:
 production-requirements:
 	pip3 install -r requirements.txt
 
-test.install_elasticsearch:
```
Comment (estute) [inline Makefile:18]: Can you update README to reflect the changes in this Makefile?

Comment (Golub-Sergey) [reply Makefile:18]: Done.

File: README.rst [hunk 3]
```diff
@@ -32,19 +32,24 @@ Getting Started
...
-   can install elasticsearch locally:
+   ``ELASTICSEARCH_LEARNERS_INDEX`` django settings. To run tests locally use:
 
    ::
 
-      $ make test.install_elasticsearch
+      $ make test
 
-   Elasticsearch requires `JDK 1.8`_; install that if you don't have it or have a different version. To run the cluster for testing:
+   For testing is used docker Elasticsearch; to run ES container:
```
Comment (estute) [inline README.rst:41]: `make test` will call `make test.run_elasticsearch` right? Could you rephrase it to something like: "This will launch an Elasticsearch Docker container for the tests. To simply run the container without starting the tests, run:"? Other than that, it looks like it's good to go

Comment (Golub-Sergey) [reply README.rst]: thanks)

**Comment by Zacharis278** [comment]
closing since this has been picked up by #444
